Public/Standalone/Set-PDFForm.ps1

```diff
@@ -1,32 +1,62 @@
-        [string] $DestinationFilePath,
-        [System.Collections.IDictionary]$FieldNameAndValueHashTable
+        [Parameter(Mandatory=$true)]$SourceFilePath,
+        [Parameter(Mandatory=$true)]$DestinationFilePath,
+        [System.Collections.IDictionary]$FieldNameAndValueHashTable,
+        [Switch] $Flatten
     )
 
-    $Script:Reader = [iText.Kernel.Pdf.PdfReader]::new($SourceFilePath)
-    $Script:Writer = [iText.Kernel.Pdf.PdfWriter]::new($DestinationFilePath)
-    $PDF = [iText.Kernel.Pdf.PdfDocument]::new($Script:Reader, $Script:Writer)
+    if ((Test-Path -LiteralPath $SourceFilePath) -and (Test-Path -LiteralPath $SourceFilePath)) {
 
-    $PDFAcroForm = [iText.Forms.PdfAcroForm]::getAcroForm($PDF, $true)
-    foreach ($Key in $FieldNameAndValueHashTable.Keys) {
-        $FormField = $PDFAcroForm.getField($Key)
-        if ( -not $FormField) { 
-            Write-Warning "No form field with name $Key found" 
-            continue
+        try{
+        $Script:Reader = [iText.Kernel.Pdf.PdfReader]::new($SourceFilePath)
+        $Script:Writer = [iText.Kernel.Pdf.PdfWriter]::new($DestinationFilePath)
+        $PDF = [iText.Kernel.Pdf.PdfDocument]::new($Script:Reader, $Script:Writer)
+        } catch {
+            $ErrorMessage = $_.Exception.Message
+            Write-Warning "Set-PDFForm - Error has occured: $ErrorMessage"
         }
+
+        $PDFAcroForm = [iText.Forms.PdfAcroForm]::getAcroForm($PDF, $true)
+        foreach ($Key in $FieldNameAndValueHashTable.Keys) {
+            $FormField = $PDFAcroForm.getField($Key)
+            if ( -not $FormField) { 
+                Write-Warning "No form field with name $Key found" 
+                continue
+            }
         
-        if ($FormField.GetType().Name -match "Button") {
-            $FormField.setValue(
-                $FormField.GetAppearanceStates()[
+            if ($FormField.GetType().Name -match "Button") {
+                $FormField.setValue(
+                    $FormField.GetAppearanceStates()[
                     [Int]$FieldNameAndValueHashTable[$Key]
-                ]
-            ) | Out-Null
-        } else {
-            $FormField.setValue($FieldNameAndValueHashTable[$Key]) | Out-Null
+                    ]
+                ) | Out-Null
+            } else {
+                $FormField.setValue($FieldNameAndValueHashTable[$Key]) | Out-Null
+            }
+        }
+
+        if($Flatten)
+        {
+            $PDFAcroForm.flattenFields()
+        }
+
+        try{
+        $PDF.Close()
+        } catch {
+            $ErrorMessage = $_.Exception.Message
+            Write-Warning "Set-PDFForm - Error has occured: $ErrorMessage"
         }
     }
-    $PDF.Close()
-}
\ No newline at end of file
+    else {
+        if(-not (Test-Path -LiteralPath $SourceFilePath)){
+            Write-Warning "Set-PDFForm - Path $SourceFilePath doesn't exists. Terminating."
+            }
+
+         if(-not (Test-Path -LiteralPath $DestinationFilePath)){
```

Comment:
There is no sense to check for existence of Destination Path as it's possible it won't exists, as we want to create new one right?